src/parser.ts

```diff
@@ -22,8 +33,72 @@ export class Parser {
         }
     }
 
-    public async parseTemplate(template: string): Promise<string> {
-        const compiledTemplate = Handlebars.compile(template);
-        return compiledTemplate(this.getDefaultContext());
+    private mapUserResponseToVariables(variables: Record<string, string>, response: Record<string, string>) {
+        const variableValues = {};
+
+        Object.keys(response).map(variable => {
+            const variableType = variables[variable].trim();
+            const variableResponse = response[variable];
+
+            if (variableType == "boolean") {
+                variableValues[variable] = variableResponse === "true";
+                return;
+            }
+
+            if (variableType == "number") {
+                variableValues[variable] = Number.parseFloat(variableResponse);
+                return;
+            }
+
+            variableValues[variable] = variableResponse;
+        });
+
+        return variableValues;
+    }
+
+    private async getVariableInputs(title: string, variables: Record<string, string>) {
+        if (Object.keys(variables).length == 0) {
+            return {};
+        }
+
+        await setTemplateVariablesView(this.dialog, title, variables);
+        const dialogResponse = (await joplin.views.dialogs.open(this.dialog));
+
+        if (dialogResponse.id === "cancel") {
+            return null;
+        }
+
+        const userResponse = dialogResponse.formData.variables;
+        return this.mapUserResponseToVariables(variables, userResponse);
+    }
+
+    public async parseTemplate(template: Note | null): Promise<string> {
+        if (!template) {
+            return "";
+        }
+
+        try {
+            const processedTemplate = frontmatter(template.body);
+            const templateVariables = processedTemplate.attributes;
+
+            const variableInputs = await this.getVariableInputs(template.title, templateVariables);
+            if (variableInputs === null) {
+                return "";
+            }
+
+            const context = {
+                ...this.getDefaultContext(),
+                ...variableInputs
+            };
+
+            const templateBody = processedTemplate.body;
+            const compiledTemplate = Handlebars.compile(templateBody);
+
+            return compiledTemplate(context);
+        } catch (err) {
+            console.error("Error in parsing template.", err);
+            await joplin.views.dialogs.showMessageBox("There was some error parsing this template.");
```

Comment:
Try to make this something actionable for the user. For example
```
"There was an error parsing this template, please review it and try again."
```

Reply:
done